src/link/language-selector.js

```diff
@@ -0,0 +1,125 @@
+import { TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { getLanguages } from '@wikimedia/language-data';
+import { isLanguageWithWiki, defaultLanguages } from './languages';
+
+export const LanguageSelector = ( { setLanguageSelector, setLang, title } ) => {
+	const [ value, setValue ] = useState( '' );
+	const [ items, setItems ] = useState( [] );
+	const limit = defaultLanguages.length;
+	const languages = getLanguages();
+
+	const normalize = ( result, language ) => {
+		if ( isLanguageWithWiki( language ) ) {
+			result.push( {
+				name: languages[ language ][ 2 ],
+				code: language,
+			} );
+		}
+		return result;
+	};
+
+	const filterLanguages = ( targetLang ) => {
+		setValue( targetLang );
+
+		if ( targetLang === '' ) {
+			setItems( defaultFilter() );
+			return;
+		}
+
+		const filtered = Object.keys( languages )
+			.filter( ( language ) => {
+				if ( languages[ language ].length > 2 ) {
+					return (
+						languages[ language ][ 2 ]
+							.toLowerCase()
+							.indexOf( targetLang.toLowerCase() ) !== -1
+					);
+				}
+				return false;
+			} )
+			.reduce(
+				( result, language ) => normalize( result, language ),
+				[]
+			);
+
+		setItems( filtered.slice( 0, limit ) );
+	};
+
+	const defaultFilter = () => {
+		const filtered = Object.keys( languages )
+			.filter( ( language ) => {
+				return defaultLanguages.indexOf( language ) !== -1;
+			} )
+			.reduce(
+				( result, language ) => normalize( result, language ),
+				[]
+			);
+
+		return filtered;
+	};
+
+	const selectLanguage = ( e ) => {
+		const languageCode = e.target.attributes[ 'data-code' ].nodeValue;
+		setLang( languageCode );
+		setLanguageSelector( false );
+	};
+
+	useEffect( () => {
+		setItems( defaultFilter() );
+	}, [] );
+
+	return (
+		<div
+			className={ `wikipediapreview-edit-inline-language-selector ${
+				! title ? `no-title` : ''
+			}` }
+		>
+			<div className="wikipediapreview-edit-inline-language-selector-header">
+				<div>{ __( 'Languages', 'wikipedia-preview' ) }</div>
+				<div
+					className="wikipediapreview-edit-inline-language-selector-header-close"
```

Comment:
Getting two X in the mobile interfac
<img src="https://user-images.githubusercontent.com/2560096/137640981-4b745d15-8533-4e10-bc16-7d218e4494cd.png" width="300" />